src/os/bluestore/BlueStore.h

```diff
@@ -621,7 +632,16 @@ class BlueStore : public ObjectStore,
 
     void dump(ceph::Formatter* f) const;
     friend std::ostream& operator<<(std::ostream& out, const Blob &b);
-
+    struct printer : public BlueStore::printer {
+      const Blob& blob;
+      uint8_t mode;
+      printer(const Blob& blob, uint8_t mode)
+      :blob(blob), mode(mode) {}
+    };
+    friend std::ostream& operator<<(std::ostream& out, const printer &p);
+    printer print(uint8_t mode) const {
+      return printer(*this, mode);
+    }
```

Comment:
shouldn't the printer be a singleton? If we add to many prints it will instantiate a printer every time we print something.